com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/ZigBeeNetworkManager.java

```diff
@@ -1767,24 +1766,23 @@ public void run() {
             return null;
         }
 
-        if (node.getNodeDescriptor() != null && networkNodes.get(node.getIeeeAddress()) != null && Objects
-                .equals(networkNodes.get(node.getIeeeAddress()).getNodeDescriptor(), node.getNodeDescriptor())) {
+        if (currentNode.getNodeDescriptor() != null) {
```

Comment:
This change will mean that the transport layer `setNodeDescriptor()` method gets called every time `refreshNode`  is called. This check should at least check that `node.getNodeDescriptor()` is not null - ie the node descriptor is being updated by the call to `refreshNode`.